ast/src/main/java/com/graphicsfuzz/common/typing/TyperHelper.java

```diff
@@ -297,49 +297,10 @@ public static Type resolveTypeOfMul(Type lhsType, Type rhsType) {
       ShadingLanguageVersion shadingLanguageVersion) {
     Map<String, List<FunctionPrototype>> builtinsForVersion = new HashMap<>();
 
-    // Trigonometric Functions
-    {
-      final String name = "sin";
-      for (Type t : genType()) {
-        addBuiltin(builtinsForVersion, name, t, t);
-      }
-    }
-    {
-      final String name = "cos";
-      for (Type t : genType()) {
-        addBuiltin(builtinsForVersion, name, t, t);
-      }
-    }
-    {
-      final String name = "asin";
-      for (Type t : genType()) {
-        addBuiltin(builtinsForVersion, name, t, t);
-      }
-    }
-    {
-      final String name = "acos";
-      for (Type t : genType()) {
-        addBuiltin(builtinsForVersion, name, t, t);
-      }
-    }
-    {
-      final String name = "tan";
-      for (Type t : genType()) {
-        addBuiltin(builtinsForVersion, name, t, t);
-      }
-    }
-    {
-      final String name = "atan";
-      for (Type t : genType()) {
-        addBuiltin(builtinsForVersion, name, t, t);
-      }
-    }
-    {
-      final String name = "atan";
-      for (Type t : genType()) {
-        addBuiltin(builtinsForVersion, name, t, t, t);
-      }
-    }
+    // 8.1: Angle and Trigonometric Functions
+
```

Comment:
@abelbriggs1 has taken the approach of checking here whether the relevant functions are supported; you have taken the approach of checking inside the "getBuiltinsForGlslVersionKINDOFFUCTIONS(...)" method.

It's a minor point but it would be good to be symmetric here.

I guess you did it inside because there a couple of different kinds of functions that came in different GLSL versions?  That seems like a good reason to do the checks inside.

@abelbriggs1 in that case perhaps in this PR @jiradeto should move the checks for your builtins inside the relevant "getBuiltinsForGlslVersionKINDOFFUNCTIONS(...)" method - what do you think?

Reply:
That's probably a good idea. I was originally leaning towards any all-encompassing checks being outside and any subsets being inside the function (see integer functions), but it'll look rather unsightly by the end. It'd be better for each built-in adding function to handle the support checks.

Reply:
Thanks - can you do that please @jiradeto ?

Reply:
Sure, fixed!